library/alloc/src/collections/btree/node.rs

```diff
@@ -807,19 +807,25 @@ impl<BorrowType, K, V, NodeType> Handle<NodeRef<BorrowType, K, V, NodeType>, mar
     }
 }
 
+impl<BorrowType, K, V, NodeType> PartialEq for NodeRef<BorrowType, K, V, NodeType> {
+    fn eq(&self, other: &Self) -> bool {
+        self.node == other.node
+    }
+}
+
```

Comment:
Apparently required to compare handles with `.ascend() == Ok(edge)` as opposed to the `.ascend().ok().unwrap() == edge` before.

Reply:
This is currently just checking for pointer equality -- which seems reasonable for testing -- but does not check the height is the same. Can we check the height too?

I am also wondering if we could add a cfg(test) on this impl.

Reply:
Oh, I get it. We can leave the old `PartialOrd` and `PartialEq` alone, and just add in the test code:
```
impl<BorrowType, K, V, NodeType> PartialEq for NodeRef<BorrowType, K, V, NodeType> {
    fn eq(&self, _other: &Self) -> bool {
        unreachable!("Phony implementation to shut up compiler")
    }
}
```
to get that `.ascend() == Ok(edge)` working.

But as to checking the height field, it seems wasteful to write that both in `PartialOrd` and `PartialEq`, as opposed to sharing a common function (which then happens to fulfill PartialEq for NodeRef anyway).

Reply:
I don't think that's a good idea. I don't think .ok().unwrap() is that bad, let's just keep that -- it seems like this is a pain otherwise.

Reply:
It doesn't seem a good idea at all, more like a workaround for a compiler bug, or more likely for something in node.rs that isn't wholly correctly specified, which I'm trying to reproduce.

But regardless of that, if you like checking the height field, and I agree, then it seems most sensible to me to do that in a `PartialEq` for `NodeRef` anyway, and the need for .ok().unwrap() happens to disappear.

Or alternatively, don't implement any `PartialEq`: replace the body of the existing `PartialEq` for `Handle` with `unreachable` (or the hardcore intrinsics alternative), because it was dead code, just recently being used by tests only.

Reply:
> it seems most sensible to me to do that in a PartialEq for NodeRef 

Or some named method like `NodeRef::is_equal`.

Reply:
If Rust's design requires `PartialEq` for `PartialOrd`, even though it isn't used for real, I suppose it's best to implement `PartialEq` for `Handle` properly, and to test it. Let's say the implementation is passable but not quite proper without fully enumerating the fields. Testing is worse: it was absent until #76983 and still only covers handles that stem from the same tree and that are supposed to be equal.

In any case, all that is quite unrelated to the subject of this particular PR, so I left it out entirely.

Reply:
> more like a workaround for a compiler bug, or more likely for something in node.rs that isn't wholly correctly specified

Actually, it was darn obvious: the Err type of the Result compared _is_ a NodeRef. :man_facepalming: So I opened up this can of worms for nothing,